kustomize/commands/edit/set/setgeneratoroptions.go

```diff
@@ -0,0 +1,101 @@
+package set
+
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/util"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type setGeneratorOptions struct {
+	GeneratorOptions types.GeneratorOptions
+	Labels           []string
+	Annotations      []string
+}
+
+func newCmdSetGeneratorOptions(fSys filesys.FileSystem) *cobra.Command {
+	var o setGeneratorOptions
+	cmd := &cobra.Command{
+		Use:   "generatoroptions",
+		Short: "Set a generatoroptions to the kustomization file",
+		Long:  "",
+		Example: `
+	# Set a generatoroptions with a "immutable = true" to the kustomization file.
+	kustomize edit set generatoroptions --immutable
```

Comment:
It seems that there should be a way to unset this (and likewise with disableNameSuffixHash, annotations, and labels). I can see two options here:

1. Change this to a string flag, and have the user pass in `--immutable=true` or `--immutable=false`.
2. Supply a new command `kustomize edit remove generatoroptions`, where we can remove the `immutable = true` with the command `kustomize edit remove generatoroptions --immutable`. 

I slightly prefer option (2), wdyt? If you agree, could you please file an issue that we need to add this new command `kustomize edit remove generatoroptions`? It doesn't have to block this PR; we can add the new command in a followup.